backend/inspirehep/pidstore/minters/bai.py

```diff
@@ -4,14 +4,52 @@
 #
 # inspirehep is free software; you can redistribute it and/or modify it under
 # the terms of the MIT License; see LICENSE file for more details.
-
+from flask import current_app
 from inspire_utils.record import get_values_for_schema
 
 from inspirehep.pidstore.minters.base import Minter
+from inspirehep.pidstore.providers import InspireExternalIdProvider
+from inspirehep.pidstore.providers.bai import InspireBAIProvider
 
 
 class BAIMinter(Minter):
     pid_type = "bai"
+    provider = InspireBAIProvider
+
+    def __init__(self, object_uuid, data):
+        super().__init__(object_uuid, data)
+        if not current_app.config.get("FEATURE_FLAG_ENABLE_BAI_PROVIDER", False):
+            self.provider = InspireExternalIdProvider
+
+    @classmethod
+    def update(cls, object_uuid, data):
+        minter = super().update(object_uuid, data)
```

Comment:
shall we call update before checking the `FEATURE_FLAG`?

Reply:
Yes, to still have previous behavior of BAI. Feature flag is checked only on minter creation, and it decides which Provider will be used (new one or previous one)

Reply:
So when flag will be turned off it will actually do nothing as ExternalProvider is using default update method which is empty.